src/planscape/forsys/merge_polygons_test.py

```diff
@@ -0,0 +1,163 @@
+import numpy as np
+
+from forsys.merge_polygons import merge_polygons
+from django.test import TestCase
+from django.contrib.gis.geos import Polygon
+from planscape import settings
+
+
+class MergePolygonsTest(TestCase):
+    def setUp(self):
+        self.xorig = -2116971
+        self.yorig = 2100954
+
+        self.side = 300
+        self.xscale = self.side
+        self.yscale = -self.side
+
+        self.srid = settings.CRS_FOR_RASTERS
+
+        self.margin = np.sqrt(2) * self.side / 2
+
+    def test_merges_contiguous_polygons(self) -> None:
+        p1 = self._create_polygon(((0, 0), (0, 5), (5, 5), (5, 0), (0, 0)))
+        # p2 is contiguous with p1.
+        p2 = self._create_polygon(((5, 0), (5, 3), (8, 3), (8, 0), (5, 0)))
+        poly = merge_polygons([p1, p2], self.margin)
+        self.assertEqual(poly.srid, self.srid)
+        self.assertEqual(poly.geom_type, "Polygon")
+        self.assertEqual(len(poly.coords), 1)
+        self.assertEqual(self._coords_to_pixel_coords(poly.coords[0]),
+                         ((5, 5), (0, 5), (0, 0), (8, 0),
+                          (8, 3), (5, 3), (5, 5)))
+
+    def test_merges_overlapping_polygons(self) -> None:
+        p1 = self._create_polygon(((0, 0), (0, 5), (5, 5), (5, 0), (0, 0)))
+        # p2 overlaps with p1.
+        p2 = self._create_polygon(((4, 0), (4, 3), (8, 3), (8, 0), (4, 0)))
+        poly = merge_polygons([p1, p2], self.margin)
+        self.assertEqual(poly.srid, self.srid)
+        self.assertEqual(poly.geom_type, "Polygon")
+        self.assertEqual(len(poly.coords), 1)
+        self.assertEqual(self._coords_to_pixel_coords(poly.coords[0]),
+                         ((5, 5), (0, 5), (0, 0), (8, 0),
+                          (8, 3), (5, 3), (5, 5)))
+
+    def test_merges_three_nonoverlapping_polygons(self) -> None:
+        p1 = self._create_polygon(((0, 0), (0, 5), (5, 5), (5, 0), (0, 0)))
+        # p2 is not contiguous or overlapping with p1.
+        p2 = self._create_polygon(((6, 0), (6, 3), (8, 3), (8, 0), (6, 0)))
+        # p3 is contiguous with p1 and not contiguous or overlapping with p2.
+        p3 = self._create_polygon(((0, 0), (0, 1), (-1, 1), (-1, 0), (0, 0)))
+        poly = merge_polygons([p1, p2, p3], self.margin)
+        self.assertEqual(poly.srid, self.srid)
+        self.assertEqual(poly.geom_type, "MultiPolygon")
+        self.assertEqual(len(poly.coords), 2)
+        self.assertEqual(self._coords_to_pixel_coords(poly.coords[0][0]),
```

Comment:
could you add word descriptions of the overlap, similar to what you did above for p2 and p3?

Reply:
Done.